modules/docs/arrow-docs/docs/docs/arrow/typeclasses/alternative/README.md

```diff
@@ -11,6 +11,60 @@ redirect_from:
 {:.intermediate}
 intermediate
 
+We use [`Option`]({{ '/docs/arrow/core/option' }}) to indicate a computation can fail somehow (that is, it can have either zero results or one result), and we use lists for computations that can have many possible results (ranging from zero to arbitrarily many results). In both of these cases, one useful operation is combining all possible results from multiple computations into a single computation. The `Alternative` type class captures this combination.
+
+`Alternative` is for [`Applicative`]({{ '/docs/arrow/typeclasses/applicative' }}) functors which also have a [`Monoid`]({{ '/docs/arrow/typeclasses/monoid' }}) structure.
+
+### Main Combinators
+
+#### Kind<F, A>.orElse
+
+Is a binary function which represents a choice between alternatives.
+
+`fun <A> Kind<F, A>.orElse(b: Kind<F, A>): Kind<F, A>`
+
+```kotlin:ank
+import arrow.core.Option
+import arrow.core.extensions.option.monadCombine.monadCombine
+
+Option.monadCombine().run {
+    val x = Option.just(1)
+    val y = Option.empty()
+    x.orElse(y)
+}
+```
+
+```kotlin:ank
+import arrow.core.ListK
+import arrow.core.extensions.listk.monadCombine.monadCombine
+import arrow.core.k
+
+ListK.monadCombine().run {
+    val x = listOf(1, 2).k()
+    val y = listOf(3, 4).k()
+    x.orElse(y)
+}
+```
+
+#### Kind<F, A>.alt
+
+It is just an infix alias over `orElse`.
+
+```kotlin:ank
+import arrow.core.Option
+import arrow.core.extensions.option.monadCombine.monadCombine
+
+Option.monadCombine().run {
+    val x = Option.just(1)
+    val y = Option.empty()
+    x alt y
+}
+```
+
+### Laws
```

Comment:
I'd add `some` and `many` but not as a hard requirement.

Also I've just realized `some` may clash with Option's constructor. Oh well.

Reply:
Yup, didn't realize earlier. Anyway, you need to be in the scope of `Alternative` to have access to this `some` and it will only appear as a function for the object. `some` as an Option constructor won't be suggested by the IDE in that situation 🤔.